plugins/labtests/models.py

```diff
@@ -0,0 +1,204 @@
+            "test_name",
+            "lab_number",
+        ]
+        result = {}
+        for field in fields:
+            result[field] = getattr(self, field)
+
+        result["observations"] = [
+            i.to_dict() for i in self.observation_set.all()
+        ]
+        return result
+
+    @classmethod
+    def get_relevant_tests(self, patient):
+        relevent_tests = [
+            "C REACTIVE PROTEIN",
+            "FULL BLOOD COUNT",
+            "UREA AND ELECTROLYTES",
+            "LIVER FUNCTION",
+            "LIVER PROFILE",
+            "GENTAMICIN LEVEL",
+            "CLOTTING SCREEN"
+        ]
+        three_weeks_ago = timezone.now() - datetime.timedelta(3*7)
+        qs = LabTest.objects.filter(
+            patient=patient,
+            datetime_ordered__gt=three_weeks_ago
+        ).order_by("datetime_ordered")
+        return [i for i in qs if i.extras.get("test_name") in relevent_tests]
+
+    def dict_for_view(self, user):
+        result = self.extras
+        result["datetime_ordered"] = self.datetime_ordered
+        result["extras"] = self.extras
+        return result
+
+
+class Observation(models.Model):
+    # as created in the upstream db
+    last_updated = models.DateTimeField(blank=True, null=True)
+    observation_datetime = models.DateTimeField(blank=True, null=True)
+    observation_name = models.CharField(max_length=256, blank=True, null=True)
+    observation_number = models.CharField(max_length=256, blank=True, null=True)
+    observation_value = models.TextField(blank=True)
+    reference_range = models.CharField(max_length=256, blank=True, null=True)
+    units = models.CharField(max_length=256, blank=True, null=True)
+    test = models.ForeignKey(LabTest, on_delete=models.CASCADE)
+
+    # as defined by us
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def create(self, observation_dict):
+        self.last_updated = serialization.deserialize_datetime(
+            observation_dict["last_updated"]
+        )
+        self.observation_datetime = serialization.deserialize_datetime(
+            observation_dict["observation_datetime"]
+        )
+        fields = [
+            "observation_number",
+            "observation_name",
+            "observation_value",
+            "reference_range",
+            "units"
+        ]
+        for f in fields:
+            setattr(self, f, observation_dict.get(f))
+        self.save()
+        return self
+
+    def to_dict(self):
+        as_dict = {}
+        as_dict["last_updated"] = format_dt(self.last_updated)
+        as_dict["observation_datetime"] = format_dt(self.observation_datetime)
+        fields = [
+            "observation_number",
+            "observation_name",
+            "observation_value",
+            "reference_range",
+            "units"
+        ]
+        for field in fields:
+            as_dict[field] = getattr(self, field)
+        return as_dict
+
+
+def create_from_old_tests(patient):
```

Comment:
I think this whole thing could be lost and make the management command iterate over upstream tests?

Reply:
resolved on 83b691d6